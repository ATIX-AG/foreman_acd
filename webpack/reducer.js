import { combineReducers } from 'redux';
import {
  cloneDeep,
  findIndex,
  findLastIndex,
} from 'lodash';

import parameterSelectionParameters from './components/ParameterSelection/ParameterSelectionReducer';
import applicationDefinitionConf from './components/ApplicationDefinition/ApplicationDefinitionReducer';
import applicationInstanceConf from './components/ApplicationInstance/ApplicationInstanceReducer';

import {
  APPLICATION_DEFINITION_PARAMETER_SELECTION_MODAL_CLOSE,
} from './components/ApplicationDefinition/ApplicationDefinitionConstants';

import {
  APPLICATION_INSTANCE_PARAMETER_SELECTION_MODAL_CLOSE,
} from './components/ApplicationInstance/ApplicationInstanceConstants';

const rootReducer = (state = {}, action: Action) => {

  let param_state = parameterSelectionParameters(state.parameterSelectionParameters, action);

  if (action.type == APPLICATION_DEFINITION_PARAMETER_SELECTION_MODAL_CLOSE) {
    action.payload.serviceParameterSelection = param_state.parameters;
  }
  let app_def_state = applicationDefinitionConf(state.applicationDefinitionConf, action);

  if (action.type == APPLICATION_INSTANCE_PARAMETER_SELECTION_MODAL_CLOSE) {
    action.payload.hostParameterSelection = param_state.parameters;
  }
  let app_ins_state = applicationInstanceConf(state.applicationInstanceConf, action);

  return {
    applicationDefinitionConf: app_def_state,
    applicationInstanceConf: app_ins_state,
    parameterSelectionParameters: param_state,
  };
};

export default rootReducer;
