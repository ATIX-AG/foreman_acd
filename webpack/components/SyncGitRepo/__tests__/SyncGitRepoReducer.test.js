import { testReducerSnapshotWithFixtures } from 'react-redux-test-utils';
import reducer, { initialState } from '../SyncGitRepoReducer';

import {
  successState,
  initSyncGitRepoPayload,
  loadPathPayload,
  loadGitCommitPayload,
  loadScmTypePayload,
  loadGitUrlPayload,
  handleGitRepoSyncRequestPayload,
  handleGitRepoSyncSuccessPayload,
  handleGitRepoSyncFailurePayload,
} from '../__fixtures__/syncGitRepoReducer.fixtures';

import {
  SYNC_GIT_REPO_INIT,
  SYNC_GIT_REPO_LOAD_SCM_TYPE,
  SYNC_GIT_REPO_LOAD_PATH,
  SYNC_GIT_REPO_LOAD_GIT_COMMIT,
  SYNC_GIT_REPO_LOAD_GIT_URL,
  SYNC_GIT_REPO_REQUEST,
  SYNC_GIT_REPO_FAILURE,
  SYNC_GIT_REPO_SUCCESS,
} from '../SyncGitRepoConstants';

const fixtures = {
  'should return initial state': {
    state: initialState,
    action: {
      type: undefined,
      payload: {},
    },
  },

  'should initialize component': {
    state: initialState,
    action: {
      type: SYNC_GIT_REPO_INIT,
      payload: initSyncGitRepoPayload,
    },
  },
  'should load path': {
    state: successState,
    action: {
      type: SYNC_GIT_REPO_LOAD_PATH,
      path: loadPathPayload,
    },
  },
  'should load git commit': {
    state: successState,
    action: {
      type: SYNC_GIT_REPO_LOAD_GIT_COMMIT,
      gitCommit: loadGitCommitPayload,
    },
  },
  'should load git url': {
    state: successState,
    action: {
      type: SYNC_GIT_REPO_LOAD_GIT_URL,
      gitCommit: loadGitUrlPayload,
    },
  },
  'should load scm_type': {
    state: successState,
    action: {
      type: SYNC_GIT_REPO_LOAD_SCM_TYPE,
      scmType: loadScmTypePayload,
    },
  },
  'should request handle git repo sync': {
    state: successState,
    action: {
      type: SYNC_GIT_REPO_REQUEST,
      payload: handleGitRepoSyncRequestPayload,
    },
  },
  'should handle git repo sync be successful': {
    state: successState,
    action: {
      type: SYNC_GIT_REPO_SUCCESS,
      payload: handleGitRepoSyncSuccessPayload,
    },
  },
  'should handle git repo sync be erroneous': {
    state: successState,
    action: {
      type: SYNC_GIT_REPO_FAILURE,
      payload: handleGitRepoSyncFailurePayload,
    },
  },
};

describe('SyncGitRepoReducer', () =>
  testReducerSnapshotWithFixtures(reducer, fixtures));
