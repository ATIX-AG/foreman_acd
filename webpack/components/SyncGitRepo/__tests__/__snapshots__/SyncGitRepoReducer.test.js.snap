// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SyncGitRepoReducer should handle git repo sync be erroneous 1`] = `
Object {
  "error": "Something really bad happend",
  "gitCommit": "master",
  "gitUrl": "https://github.com/alpha.git",
  "loading": false,
  "path": "https://example.com",
  "scmType": Object {
    "git": "git",
  },
  "scmTypes": Object {
    "dictionary": "dictionary",
    "git": "git",
  },
}
`;

exports[`SyncGitRepoReducer should handle git repo sync be successful 1`] = `
Object {
  "gitCommit": "master",
  "gitUrl": "https://github.com/alpha.git",
  "path": "https://example.com",
  "scmType": Object {
    "git": "git",
  },
  "scmTypes": Object {
    "dictionary": "dictionary",
    "git": "git",
  },
}
`;

exports[`SyncGitRepoReducer should initialize component 1`] = `
Object {
  "gitCommit": "master",
  "gitUrl": "https://github.com/alpha.git",
  "loading": false,
  "path": "https://example.com",
  "scmType": Object {
    "git": "git",
  },
  "scmTypes": Object {
    "dictionary": "dictionary",
    "git": "git",
  },
}
`;

exports[`SyncGitRepoReducer should load git commit 1`] = `
Object {
  "gitCommit": Object {
    "gitCommit": "master",
  },
  "gitUrl": "https://github.com/alpha.git",
  "path": "https://example.com",
  "scmType": Object {
    "git": "git",
  },
  "scmTypes": Object {
    "dictionary": "dictionary",
    "git": "git",
  },
}
`;

exports[`SyncGitRepoReducer should load git url 1`] = `
Object {
  "gitCommit": "master",
  "gitUrl": undefined,
  "path": "https://example.com",
  "scmType": Object {
    "git": "git",
  },
  "scmTypes": Object {
    "dictionary": "dictionary",
    "git": "git",
  },
}
`;

exports[`SyncGitRepoReducer should load path 1`] = `
Object {
  "gitCommit": "master",
  "gitUrl": "https://github.com/alpha.git",
  "path": Object {
    "path": "https://example.com",
  },
  "scmType": Object {
    "git": "git",
  },
  "scmTypes": Object {
    "dictionary": "dictionary",
    "git": "git",
  },
}
`;

exports[`SyncGitRepoReducer should load scm_type 1`] = `
Object {
  "gitCommit": "master",
  "gitUrl": "https://github.com/alpha.git",
  "path": "https://example.com",
  "scmType": Object {
    "scmType": Object {
      "git": "git",
    },
  },
  "scmTypes": Object {
    "dictionary": "dictionary",
    "git": "git",
  },
}
`;

exports[`SyncGitRepoReducer should request handle git repo sync 1`] = `
Object {
  "gitCommit": "master",
  "gitUrl": "https://github.com/alpha.git",
  "loading": true,
  "path": "https://example.com",
  "scmType": Object {
    "git": "git",
  },
  "scmTypes": Object {
    "dictionary": "dictionary",
    "git": "git",
  },
}
`;

exports[`SyncGitRepoReducer should return initial state 1`] = `
Object {
  "loading": false,
}
`;
